test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -154,21 +271,51 @@ public void TwoConnectionsWithFailure()
                 conn1.EnlistTransaction(Transaction.Current);
                 conn2.EnlistTransaction(Transaction.Current);
 
-                Assert.That(conn1.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test1')"), Is.EqualTo(1));
-                Assert.That(conn2.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test2')"), Is.EqualTo(1));
+                Assert.That(conn1.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test1')"), Is.EqualTo(1),
+                    "Unexpected first insert rowcount");
+                Assert.That(conn2.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test2')"), Is.EqualTo(1),
+                    "Unexpected second insert rowcount");
+
                 conn1.ExecuteNonQuery($"SELECT pg_terminate_backend({conn2.ProcessID})");
                 scope.Complete();
                 Assert.That(() => scope.Dispose(), Throws.Exception.TypeOf<TransactionAbortedException>());
+
                 AssertNoPreparedTransactions();
                 using (var tx = conn1.BeginTransaction())
                 {
-                    Assert.That(conn1.ExecuteScalar(@"SELECT COUNT(*) FROM data"), Is.EqualTo(0));
+                    Assert.That(conn1.ExecuteScalar(@"SELECT COUNT(*) FROM data"), Is.EqualTo(0),
+                        "Unexpected data count");
                     tx.Rollback();
                 }
             }
         }
 
         [Test]
+        public void ConnectionReuseAfterDistributedTransaction()
```

Comment:
This test is for checking if supporting a null enlistment could be required for dodging some race condition. I should probably add a variation with a durable ressource enlisted, and some loops.